internal/decryption/keyproviders/aws.go

```diff
@@ -37,69 +37,35 @@ type AWSKeyIdentifier struct {
 	SecretAccessKey string
 	SessionToken    string
 
-	credentials *credentials.Credentials
+	cfg aws.Config
```

Comment:
credential is no longer a thing, it seems closer to how we did [gcp](https://github.com/mongodb/mongodb-atlas-cli/blob/dc9e02e34c51ad670b21d83ca97d73836034f19c/internal/decryption/keyproviders/gcp.go) so following that pattern

Reply:
the thing here is in GCP there is no way to set secrets by hand not even in their cli, you're obliged to get a secret and provide it on a config I my memory is not completely bananas

Reply:
For AWS we normally have the API KEY / API SECRET KEY to input via parameters on their CLI, I imagine that you may be able to do that on the new SDK

Reply:
looking at their SDK I've found this https://github.com/aws/aws-sdk-go-v2/blob/main/credentials/static_provider.go#L29-L53 I think we would need to check and ask for credentials

Reply:
I was trying to follow https://aws.github.io/aws-sdk-go-v2/docs/migrating/#client-construction, I think you did catch a case of provided creds as the new `config.LoadDefaultConfig` replaces the both `credentials.EnvProvider`, `credentials.SharedCredentialsProvider` and given the test was passing I called it a day, I suppose the test is then using env variables(?)

The comment here on gcp approach is mostly because for gcp we do the same on saving the conf and not the creds, and we also have the issue on the context not being passed but using the background context

Reply:
I added the static credential provider to config.LoadDefaultConfig, which hopefully addresses your comment

Reply:
👍

Reply:
this actually doesn't work the same anymore and it won't chain the providers being a last write win, seems like the new SDK doesn't support that feature anymore so holding on this PR for now

Reply:
I've been rereading this PR I think the only thing we have to do is check if the three variables are filled if so override the default, otherwise, we should commit the `credentials.SharedCredentialsProvider`

Reply:
I think this should be now working as it was before